apps/src/p5lab/drawUtils.js

```diff
@@ -116,3 +116,36 @@ export function multilineText(
   lines.forEach((line, i) => p5.text(line, x, y + i * size));
   p5.pop();
 }
+
+/**
+ * Draw a validation bar - a P5 shape comprised of a rectangle
+ * at the bottom of the canvas.
+ *
+ * @param {P5} p5
+ * @param {Number} width
+ * @param {String} state
+ * @returns {Object}
+ */
+export function validationBar(
+  p5,
+  width,
+  state,
+  {x = 0, y = 390, height = 10} = {}
```

Comment:
Should we use named constants and/or explain where these hardcoded values come from?

Reply:
I imported `APP_HEIGHT` to use with `y` here. The number 10 is somewhat arbitrary (for what "looks good"), similar to the numbers in the `speechBubble` function above. What's our rule for when it's okay to use a number and when should we use name constants?

Reply:
If there isn't a named constant, then there should at least be a comment, and oftentimes both.  The rule is to avoid "magic numbers" where it's not immediately obvious what the number represents.  In this case, if this is related to the number of vertical pixels in the playspace, then there should be a comment and/or named variable that makes that clear.